### DISCUSSION TITLE
Интерфейс и управление ZCAD

### DISCUSSION THREAD

**Discussion by gtfox** [zamtmn/zcad#132] [Ideas]
1. Командная строка частично перекрывается панелями. Возможно это из-за моего нубука с 1366х768
Закрытие панелей не помогает, сползает ниже
![Безымянный3](https://github.com/user-attachments/assets/d680c215-173f-4f6f-9449-a2ae9f1bdf17)
2. Не показываются картинки устройств
![Безымянный2](https://github.com/user-attachments/assets/5392234c-5a2c-45fa-9483-a64bc02f8d65)
3. Предложение: добавить кнопки или контекстное меню для вкладки чертежа
![Безымянный_](https://github.com/user-attachments/assets/d1fe142d-d05e-47e5-a308-6af441902213)
![Безымянный](https://github.com/user-attachments/assets/ada9544e-8b9d-4524-b4fb-7ba2d272a7e5)
4. Не нашел как сделать курсив для текста. Причем в устройстве был курсив. Я ему изменил стиль, а курсив сохранился.
![Безымянный4](https://github.com/user-attachments/assets/1553dce9-9e3f-43ea-8376-4bcc977d2368)
5. ПКМ над примитивом не выделяет его. Возможно так и должно быть... В визио выделяет.
6. Ctr+X не копирует в буфер = удаление. В офисах эта процедура позволяет вставить вырезанный объект из буфера.
7. Ctrl+D не дублирует выделенное.
8. Новый набор выбора - не добавляет в набор при зажатом Ctrl+ЛКМ. В этом режиме невозможно выделить группу объектов при зажатом Ctrl.

**Comment by zamtmn** [comment]
1. да, из-за маленького разрешения
![image](https://github.com/user-attachments/assets/e99a2dac-4c67-40c9-bf0e-a0d0de61aba4)
вариант 1 увелич ее за сплитер 
![image](https://github.com/user-attachments/assets/9ff4990a-3fa8-4b68-802b-0c4fbc2b7a08)
![image](https://github.com/user-attachments/assets/f31114b6-108b-41fb-a3ed-409f0e700760)
вариант 2 выключи ее
![image](https://github.com/user-attachments/assets/6364e712-a723-4ffb-9a64-33f37ba8dedc)
в первом варианте нужно будет м меню settings/save windows layout as default, в варианте 2 нужно settings/save options чтобы настройки сохранились о следующего сеанса зкада

2. сгенерируй их. в свежезапущеном зкаде выполни 
![image](https://github.com/user-attachments/assets/d833ea9e-9cc3-4f9a-9c16-2e8abe9c1c16)

3. надо. и они есть в линуксе. в винде это сложнее, т.к. штатный таб контрол не умеет кнопки, нужно придумывать костыль

4. есть несколько способов. если это однострочный текст проще всего этот параметр 
![image](https://github.com/user-attachments/assets/3913ea5d-9350-4d98-9c8a-bc911c877252)

5. пкм - это контекстное меню. в автокаде тоже не выделяет. наверно надо сделать действия мышки настраиваемыми

6,7 - добавлю, чтоб незабыл сделай бпгрепорт

8. не понял что за новый способ? сейчас Ctrl+ЛКМ = ЛКМ емнип. а вот Shift+ЛКМ - развыделение.

**Comment by gtfox** [comment]
> не понял что за новый способ? сейчас Ctrl+ЛКМ = ЛКМ емнип. а вот Shift+ЛКМ - развыделение.

![Анимация](https://github.com/user-attachments/assets/e0ed897f-f1c9-4975-9223-8b0aed2a8ac0)

Остальные ответы попробую позже... с наскоку не получилось :)

**Comment by zamtmn** [comment]
по умолчанию простой клик на примитиве добавляет его в выбор, выбор растет. шифт+клик убирает из выбора. есть режим когда в выборе только один объект активируется кнопкой "1" 

https://github.com/user-attachments/assets/544b8ae6-a957-4a64-a92f-a45c89f524df

**Reply by gtfox** [reply]
Да. Про него речь и идёт. Называется оно "Новый набор выбора". По мне не очень название. Что-то типа "одиночный выбор"/"групповой выбор" - хотя тоже фигня.
Такой режим по умолчанию в везде (в визио, в офисах, в splane), кроме автокада :)
При этом режиме, при зажатом Ctrl кликаем по объектам и они собираются в набор
Так вот в ZCAD такого не происходит

**Comment by gtfox** [comment]
Касаемо текстов:
Создал стиль. Oblique выставил 0,2
![Безымянный2](https://github.com/user-attachments/assets/f9d04763-efa5-4312-b68f-7276b701c30b)
Применил стиль к МТекст - работает!
Применил стиль к Текст - шрифт есть, наклона нет. Вручную выставил наклон 20. Не совпадает с наклоном МТекст-а
Белый - МТекст, Серый - Текст
![Безымянный](https://github.com/user-attachments/assets/33e54d26-4182-4e27-af5f-57e3f3ce470f)

если это баг - перекину в issues

**Comment by zamtmn** [comment]
Да это баг. 0.2 похоже в радианах))

**Comment by gtfox** [comment]
По поводу командной строки: вылазить не хочет. Ну и фик с ней :)
По поводу картинок устройств: сделал, не помогло.
```
Запущена команда:ExecuteFile
Выполняется скрипт "*components\blockpreviewexport.cmd";
```
Тоже фик с ними :)

**Comment by zamtmn** [comment]
по командной строке - хз
по картинкам глянул - точно я недавно сломал когда отказался от макроса * в путях в пользу $(zcadpath) сейчас починю

**Comment by zamtmn** [comment]
![image](https://github.com/user-attachments/assets/515f89b3-76e3-4823-ae67-694e2b30076e)

починил требует пересборки окружения, или просто подменить файл components\blockpreviewexport.cmd на новый

**Reply by gtfox** [reply]
Спасибо! Выглядит поинтереснее вопросиков :)

**Comment by gtfox** [comment]
Предложение: поменять 3 иконки

![02](https://github.com/user-attachments/assets/0a04a1a7-2e56-4979-aa07-78bdd5b2bf3a)
![01](https://github.com/user-attachments/assets/2d12f9c5-632a-48b1-ade4-596ef2f5b3bb)
[PNG.zip](https://github.com/user-attachments/files/17347552/PNG.zip)

В релизе лежат в cad\images и cad\images\actions
В исходниках лежат только svg, но конкретно этих нет svg

**Reply by zamtmn** [reply]
готово

**Comment by gtfox** [comment]
Заметил такую штуку.


https://github.com/user-attachments/assets/6f23f082-1699-447d-acb3-b50829fe5c4f

**Comment by zamtmn** [comment]
Я рассуждал так - жмеи на кнопку + = делаем в дереве всё плюсом. т.е. сворачиваем.
По мне не принципиально, легко можно поменять

**Comment by gtfox** [comment]
Предложение по интерфейсу:

Для светлой темы инспектора:
Выделять заголовки цветом и жирным шрифтом
Более светлая сетка

![001](https://github.com/user-attachments/assets/30e267e8-5f2d-4700-8e2b-16d72129ff8d)

Для тёмной темы инспектора
Выделение цветом и лёгкая сетка
Пример из автокада 2016

![003](https://github.com/user-attachments/assets/6b7b8fb3-1b22-4a03-b6aa-27d46db80076)

Для тёмной темы навигатора
Светлые линии папок

![004](https://github.com/user-attachments/assets/686101e8-ad3f-4a3b-ab2c-e5e27ad82810)

**Comment by zamtmn** [comment]
Дизайнер из меня плохой. добавил настройку цвета
![image](https://github.com/user-attachments/assets/e0ec6030-72ad-4312-bf9b-deb58351ea8c)
по линиям наверно тоже добавлю. но это всё только в old style будет скорее всего работать
по навигаторам - это внутренняя отрисовка vtv, она не расчитана на темную тему, надо репортить - патчить
